src/main/java/uk/gov/indexer/Application.java

```diff
@@ -16,7 +16,10 @@ public static void main(String[] args) throws InterruptedException {
         Configuration configuration = new Configuration(args);
         Set<String> registers = configuration.getRegisters();
 
-        ScheduledExecutorService executorService = Executors.newScheduledThreadPool((int) registers.stream().filter(r -> configuration.cloudSearchEndPoint(r).isPresent() || configuration.elasticSearchEndPoint(r).isPresent()).count() + registers.size());
+        ScheduledExecutorService executorService = Executors.newScheduledThreadPool(
+                (int) registers.stream().filter(r -> configuration.cloudSearchEndPoint(r).isPresent() || configuration.elasticSearchEndPoint(r).isPresent()).count() +
```

Comment:
is it worth extracting this stream processing into a separate method and giving it a meaningful name?